src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -392,7 +393,15 @@ public void NotifyOnPendingClusterTransaction()
         private long? _nextClusterCommand;
         private long _lastCompletedClusterTransaction;
         public long LastCompletedClusterTransaction => _lastCompletedClusterTransaction;
+        private int _numberOfGraphQueries;
+
+        public int GetNextGraphQueryId()
+        {
+            return Interlocked.Increment(ref _numberOfGraphQueries);
```

Comment:
Can we move it to GraphQueryRunner and acces it via Database.QueryRunner.Graph.CurrentlyRunningGraphQueries ?